syncer/data_fetch.go

```diff
@@ -377,23 +383,35 @@ func (d *DataFetch) receiveOpinions(ctx context.Context, req *opinionRequest, pe
 	}
 }
 
-// GetEpochATXs fetches all ATXs published in the specified epoch from a peer.
-func (d *DataFetch) GetEpochATXs(ctx context.Context, epoch types.EpochID) error {
-	peers := d.fetcher.GetPeers()
-	if len(peers) == 0 {
-		return errNoPeers
-	}
-	peer := p2p.NoPeer
+func (d *DataFetch) pickAtxPeer(epoch types.EpochID) p2p.Peer {
+	d.mu.Lock()
+	defer d.mu.Unlock()
 	if _, ok := d.atxSynced[epoch]; !ok {
 		d.atxSynced[epoch] = map[p2p.Peer]struct{}{}
 		delete(d.atxSynced, epoch-1)
 	}
+	peers := d.fetcher.GetPeers()
 	for _, p := range peers {
```

Comment:
Is it OK to hold the mutex locked for the duration of `d.fetcher.GetPeers()`? (IDK how long can it take to complete).

Reply:
thanks. the peer info is static info in p2p.
also only one sync can happen at at time. this lock is for future misuse.